run-tests.py

```diff
@@ -1,17 +1,32 @@
+from __future__ import annotations
 
 import glob
 import os
+import shutil
 import subprocess
 import sys
 
-def cleanup(out):
-  ret = ''
-  for s in out.decode('utf-8').split('\n'):
-    if len(s) > 1 and s[0] == '#':
+
+def cleanup(out: str) -> str:
+  parts = []
+  for line in out.splitlines():
+    if len(line) > 1 and line[0] == '#':
       continue
-    s = "".join(s.split())
-    ret = ret + s
-  return ret
+    parts.append("".join(line.split()))
```

Comment:
This is not supposed to be stripped. The empty lines this might remove are actually important because this is preprocessor output and that would change the locations of the code.